### ISSUE TITLE
Failed to run mypy when duplicate module names

### ISSUE THREAD

**Issue by atif93** [PyCQA/prospector#350]
**Describe the bug**
Prospector shows `Failed to run mypy` error when there's a duplicate module mypy error.
The problem stems from the way mypy messages are handled. In `1.1.6.2` the problem is because [this line](https://github.com/PyCQA/prospector/blob/1.1.6.2/prospector/tools/mypy/__init__.py#L79) expects to have 5 values returned but the error message of duplicate modules (`mypy: error / Duplicate module named 'XXXX'`) only has 3 values (line number and char number missing). Similar problem is present in `1.1.6.4` and `1.1.7` also.

**To Reproduce**
1. Have a codebase with multiple top-level modules having the same name.
2. Call prospector with mypy.
3. See error `Failed to run mypy`.

**Expected behavior**
The actual mypy error should be shown which is `mypy: error / Duplicate module named 'XXXX'`.

**Environment (please complete the following information):**
 - OS: macOS
 - Tool: mypy
 - Prospector version: 1.1.6.2, 1.1.6.3, 1.1.7
- Python version: 3.6.8

**Additional context**
Currently only solution seems to be avoiding 2 modules having the same name.

**Comment by chocoelho** [comment]
Thanks for the report @atif93. I'll be checking this in the next few days.

**Comment by hikoz** [comment]
with `mypy>=0.730`

**Comment by miecio190** [comment]
I am not sure if I faced the same error with mypy. I hope you can help me to investigate it.

Here is log produced by prospector when I tried to run mypy on certain project:
[log.txt](https://github.com/PyCQA/prospector/files/4184987/log.txt)

I use following verions of packages (let me know if there is something relevant that I have missed):
mypy==0.761
mypy-extensions==0.4.3
prospector==1.2.0

**Comment by atif93** [comment]
It's difficult to say what the actual error in your case is. You'll have to change prospector code to actually print the exception and see why it fails. That's how I had found this issue.

**Comment by carlio** [comment]
If you run it with the `--die-on-tool-error` it will probably print the exception for you without changing the code, though I don't know if this particular error was hidden somehow.

**Comment by atif93** [comment]
Yeah sorry you can run with `--die-on-tool-error` to get the exception. But for getting the actual mypy error, I had to change the code as far as I remember, because the exception only said that there were only 3 values to unpack and 5 were expected.

**Comment by miecio190** [comment]
This is the traceback that I have got (used ```--die-on-tool-error``` flag:
![image](https://user-images.githubusercontent.com/51504507/74434548-f147d680-4e62-11ea-96d1-755732ef7e56.png)

**Comment by atif93** [comment]
It's difficult to say. If you could print the variable `message` somehow by using a debugger or going to `C:\Python37\lib\site-packages\prospector\tools\mypy\__init__.py` on your computer and putting a `print(message)` in starting of `format_message` function, that would tell the actual `mypy` error.

**Comment by chocoelho** [comment]
@miecio190 your error is for the latest `prospector` and `mypy` versions, right? Just checking, as this was dealt with some time ago, but if you could provide a reproducible env it'd be helpful. Basically, `pip freeze` list and boilerplate code that would trigger this issue.

**Comment by miecio190** [comment]
@chocoelho I have got quite a lot packages installed, here is my full pip freeze:
```aenum==2.2.3
appdirs==1.4.3
asgiref==3.2.3
astroid==2.3.3
atomicwrites==1.3.0
attrs==19.3.0
bandit==1.6.2
behave==1.2.6
bitstruct==8.8.1
cached-property==1.5.1
cantools==32.20.3
certifi==2019.9.11
chardet==3.0.4
Click==7.0
colorama==0.4.3
coverage==5.0.3
coveralls==1.10.0
cycler==0.10.0
defusedxml==0.6.0
diskcache==4.1.0
Django==3.0.3
docopt==0.6.2
docutils==0.16
dodgy==0.2.1
et-xmlfile==1.0.1
fasteners==0.15
frosted==1.4.1
future==0.18.0
gitdb2==2.0.6
GitPython==3.0.7
idna==2.8
importlib-metadata==0.23
isodate==0.6.0
isort==4.3.21
jdcal==1.4.1
jenkins-job-builder==3.0.2
Jinja2==2.10.3
kiwisolver==1.1.0
lazy-object-proxy==1.4.3
lxml==4.4.1
MarkupSafe==1.1.1
matplotlib==3.1.2
mccabe==0.6.1
mock==3.0.5
mod-wsgi==4.7.0
monotonic==1.5
more-itertools==8.2.0
multi-key-dict==2.0.3
mypy==0.761
mypy-extensions==0.4.3
nose==1.3.7
numpy==1.17.2
openpyxl==3.0.0
packaging==20.1
pandas==0.25.1
parse==1.12.1
parse-type==0.5.2
pbr==5.4.3
pep8-naming==0.4.1
pies==2.6.7
plotly==4.3.0
pluggy==0.13.1
prospector==1.2.0
py==1.8.1
pycodestyle==2.4.0
pydocstyle==5.0.2
pyelftools==0.26
pyflakes==2.1.1
Pygments==2.5.2
pylint==2.4.4
pylint-celery==0.3
pylint-django==2.0.12
pylint-flask==0.6
pylint-plugin-utils==0.6
pyparsing==2.4.6
pyroma==2.6
pytest==5.3.5
pytest-django==3.8.0
python-can==3.3.2
python-dateutil==2.8.0
python-jenkins==1.5.0
pytz==2019.3
pywin32==227
PyYAML==5.1.2
requests==2.22.0
requests-toolbelt==0.9.1
requirements-detector==0.6
retrying==1.3.3
setoptconf==0.2.0
six==1.14.0
smmap2==2.0.5
snowballstemmer==2.0.0
sqlparse==0.3.0
stevedore==1.31.0
textparser==0.21.1
typed-ast==1.4.1
typing-extensions==3.7.4.1
urllib3==1.25.6
virtualenv==16.7.9
virtualenvwrapper-win==1.2.5
vulture==0.24
wcwidth==0.1.8
windows-curses==2.1.0
wrapt==1.11.2
yamlordereddictloader==0.4.0
zeep==3.4.0
zipp==0.6.0```

@atif93 here is the message:
```example\lauterbach\test.py: error: Duplicate module named 'test' (also at 'C:\Repositories\cslp-test-framework\example\can\test.py')```
So I guess I am also dealing with duplicate of module name.

Thank you guys for the support.

**Comment by atif93** [comment]
Yes, looks like the same error.
Temporary fix would be to change the names.
